controllers/cloud.redhat.com/providers/pullsecrets/provider.go

```diff
@@ -0,0 +1,21 @@
+package pullsecrets
+
+import (
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/providers"
+	core "k8s.io/api/core/v1"
+)
+
+// ProvName sets the provider name identifier
+var ProvName = "pullsecret"
+
+// CoreEnvPullSecrets is the pull_secrets for the app.
+var CoreEnvPullSecrets = providers.NewMultiResourceIdent(ProvName, "core_env_pull_secrets", &core.Secret{})
+
+// GetPullSecret returns the correct end provider.
+func GetPullSecret(c *providers.Provider) (providers.ClowderProvider, error) {
+	return NewPullSecretProvider(c)
+}
+
+func init() {
+	providers.ProvidersRegistration.Register(GetPullSecret, 98, ProvName)
```

Comment:
Should we consider refactoring this out to an enum or some layer/tier system? I know 98 is later in the process, but I don't know how much later it is exactly.

Reply:
We _could_ but then how would we do half tiers.....this is a pretty delicate piece of the system